### ISSUE TITLE
The game crashes when clicking on the loading screen

### ISSUE THREAD

**Issue by Tfarcenim** [MinecraftForge/MinecraftForge#5888]
Clicking on the game screen while it's still in startup phase crashes the game regardless of what mods are present.

https://pastebin.com/NKap2QwT

**Comment by LexManos** [comment]
Unable to reproduce, the error in your log seems to indicate that ingameGui is null which shouldn't be possible in that stack.
Are you able to reproduce? If so we need more information.

**Comment by Unnoen** [comment]
I can't reproduce either, spamming clicks anywhere on the screen while loading doesn't affect it at all for me.

**Comment by Tfarcenim** [comment]
Just reproduced it in 47 with 0 mods installed
![2019-07-10_16-28-07](https://user-images.githubusercontent.com/44327798/61002542-c24b0e80-a32f-11e9-9f4b-2c0c7694a36a.gif)

crash log
https://pastebin.com/AWYDdYiy

**Comment by JayJay1989** [comment]
can also reproduce. this is when the window is created before the loading take place.
crashlog: [link](https://gist.github.com/JayJay1989/a756d6605b8c3ec6e56dede349ccc003)

**Comment by LexManos** [comment]
Alright, wrote a macro that spammed left click every 1 MS.
This is a somewhat race condition that exists in vanilla.
Tho I am unable to find the vanilla bug report for it.
TLDR: The mouse handler is registered at the top of init(), the ingameGui is set in the middle of init(), the loadingGui is set at the end of init(), the handler tries to reference the ingameGui if the loadingGui isn't set. So if your click comes in before ingame is set, but after the handler is registered. It'll NPE.
